blog/posts/scraping-example-expenses.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Scraping behind login
+subTitle: An example in accessing account-specific information
+date: 2020-12-23
+author: Giovanni Rago
+githubUser: ragog
+tags: 
+  - scraping
+---
+
+Puppeteer and Playwright can be particularly useful when scraping data accessible only behind a login wall. This article shows a practical example of such a case.
+
+<!-- more -->
+
+## Scraping expenses on Amazon
+
+For our example, we will be logging in to our Amazon account and scraping the price off each order in the previous year, then adding them all up to show us our total Amazon expense over that period of time.
+
+A combination of UI automation and scraping will allow us to first log in to the platform, and then to retrieve the information about all our orders.
+
+:::: tabs :options="{ useUrlFragment: false }"
+::: tab Puppeteer 
+<<< @/blog/snippets/puppeteer/scraping-example-expenses.js
+:::
+::: tab Playwright
+<<< @/blog/snippets/playwright/scraping-example-expenses.js
+:::
+::::
+
+::: warning
+This example is only intended for learning purposes. Always make sure the website you are planning to scrape allows such behaviour.
```

Comment:
Not sure if worth it, but in my case, I use 2FA for amazon, should we put a warning as well?

Reply:
Totally right. I will add it here; 2FA also merits its own section in the linked page in the warning below ("Challenging automation flows")